### ISSUE TITLE
GoogleCloudPlatform/github-actions/setup-gcloud has been deprecated, please switch to google-github-actions/setup-gcloud. 

### ISSUE THREAD

**Issue by vorburger** [vorburger/www.fineract.dev#4]
While investigating #3, I've noticed that e.g. on https://github.com/apache/fineract/actions/runs/452581732 there is an annotation saying:

_Thank you for using setup-gcloud Action. GoogleCloudPlatform/github-actions/setup-gcloud has been deprecated, please switch to google-github-actions/setup-gcloud._

I'll see what needs to be done about that some time (but possibly not today). If anyone is interested in helping, please feel free to send PRs - I suspect this means the `uses: GoogleCloudPlatform/github-actions/setup-gcloud@master` in https://github.com/apache/fineract/blob/develop/.github/workflows/fineract.dev.yaml needs to be changed.

**Comment by vorburger** [comment]
It now (also or instead?) says the following, but after #8 I've had enough and will stop now:

_Deploy on Fineract.devgoogle-github-actions/setup-gcloud is pinned at HEAD. We strongly advise against pinning to "@master" as it may be unstable. Please update your GitHub Action YAML from:      uses: 'google-github-actions/setup-gcloud@master'  to:      uses: 'google-github-actions/setup-gcloud@v0'  Alternatively, you can pin to any git tag or git SHA in the repository._

_Deploy on Fineract.dev"service_account_key" has been deprecated. Please switch to using google-github-actions/auth which supports both Workload Identity Federation and Service Account Key JSON authentication. For more details, see https://github.com/google-github-actions/setup-gcloud#authorization_

@ptuomola if you to raise a PR to update https://github.com/apache/fineract/blob/develop/.github/workflows/fineract.dev.yaml for whatever it's trying to want us to do above, feel free. (But "it works" as is, so not urgent - but it sounds like it may break sooner or later.)

**Comment by ptuomola** [comment]
Hi @vorburger - if I understood correctly what this is complaining about, then https://github.com/apache/fineract/pull/2014 should fix it. But as far as I can see there's no way to test this action as it doesn't get run unless we merge it. So should I just go ahead and try it out?

Btw how did you get the secrets for GCP set up for Fineract? I thought Apache github projects did not support secrets... if we can get a secret set up for SonarQube we can get rid of the whole Travis build altogether.

**Comment by ptuomola** [comment]
@vorburger unfortunately it looks like the deployment to fineract.dev is no longer working. I thought it was because of the changes I've made to address this issue. But I've just rolled my changes back, and it is still failing - so obviously that was not the case.

Looking at the timing of the last successful build and the one after that, I think the change that broke the deployment was this one: Replace command line Docker image build with Gradle Jib plugin https://github.com/apache/fineract/commit/6e052e536be5119b68ae095b923538912fde716a

Unfortunately I can't see the logs in the Google Cloud Build Console - I just get an error saying "Error: The caller does not have permission" - so I can't see what's broken by that. Perhaps you could grant me the required permission and then I can see if I can help to fix this?

FYI @vidakovic

**Comment by vorburger** [comment]
> @vorburger unfortunately it looks like the deployment to fineract.dev is no longer working.

Let's sort that out in #11.

> Unfortunately I can't see the logs in the Google Cloud Build Console - I just get an error saying "Error: The caller does not have permission" - so I can't see what's broken by that. Perhaps you could grant me the required permission and then I can see if I can help to fix this?

Let's sort that out in #10.

That way we can keep this issue for the original goal (and you can re-revert your https://github.com/apache/fineract/pull/2014), after we fixed #10 and #11.

**Comment by vorburger** [comment]
> Btw how did you get the secrets for GCP set up for Fineract? I thought Apache github projects did not support secrets... if we can get a secret set up for SonarQube we can get rid of the whole Travis build altogether.

@ptuomola Ha! That's MAGIC... 😈 See [this slide from my ApacheCon prez](https://docs.google.com/presentation/d/1-VP4bNkc5kZ3B0yme_vYLiY1qpswnfz8ainnX5fp3l8/edit#slide=id.g96b92e9d0e_0_14) linking to [this old blog post of mine](http://blog2.vorburger.ch/2020/05/fineractdev-cicd-from-github-to-google.html), which explains how to set up a Secret on repos one only has Write but not Admin to, such as ASF. I've added a link to the README in https://github.com/vorburger/www.fineract.dev/commit/5c8020b00db738d29d21091ec10393c1817e1708. (Feel free to give me a shout if you need more about this; perhaps best in a separate issue from this one?)

**Comment by ptuomola** [comment]
Brilliant - thanks! Will give this a go a bit later, would be great to get completely rid of Travis...

**Comment by ptuomola** [comment]
This should now be resolved - I've reintroduced the patch that fixes the two actions that were marked as deprecated. 

On another note, I wonder if you still need to use the gcloud beta commands or whether the build ones are now "mainstream"?

**Comment by vorburger** [comment]
I've verified that the latest Fineract.dev related GitHub Action runs ([e.g. here](https://github.com/apache/fineract/actions/runs/1661888705)) don't show these warnings anymore, and am thus resolving this issue now.

**Comment by vorburger** [comment]
> wonder if you still need to use the gcloud beta commands or whether the build ones are now "mainstream"?

see #14
